src/main/java/org/springframework/cloud/deployer/spi/cloudfoundry/AbstractCloudFoundryDeployer.java

```diff
@@ -82,6 +83,22 @@ int memory(AppDeploymentRequest request) {
 		return (int) ByteSizeUtils.parseToMebibytes(withUnit);
 	}
 
+	int memory(AppScaleRequest request) {
+		if (request.getProperties().isPresent() && request.getProperties().get() != null) {
+			return (int) ByteSizeUtils.parseToMebibytes((String)request.getProperties().get().getOrDefault(AppDeployer.MEMORY_PROPERTY_KEY,
```

Comment:
I felt it to be weird that we added this properties map as type `<String, Object>` which is now resulting a blind casting to string. We either need to go to `<String, String>` type or properly handle Object here.